shared/src/components/enroll/confirm-join-course.cjsx

```diff
@@ -0,0 +1,81 @@
+React = require 'react'
+ReactDOM = require 'react-dom'
+BS = require 'react-bootstrap'
+ENTER = 'Enter'
+
+AsyncButton = require '../buttons/async-button'
+CcConflictMessage = require './cc-conflict-message'
+MessageList = require './message-list'
+
+ConfirmJoinCourse = React.createClass
+
+  propTypes:
+    courseEnrollmentActions: React.PropTypes.object.isRequired
+    courseEnrollmentStore: React.PropTypes.object.isRequired
+    optionalStudentId: React.PropTypes.bool
```

Comment:
Something you can do here is use `React.PropTypes.shape` to ensure that the object that's passed in has the properties you're using, such as:

```
courseEnrollmentStore: React.PropTypes.shape(
   getEnrollmentChangeId: React.PropTypes.func.isRequred
  hasConflict: React.PropTypes.func.isRequred
  description: React.PropTypes.func.isRequred
  teacherNames: React.PropTypes.func.isRequred
).isRequired,
courseEnrollmentActions: React.PropTypes.shape(
  confirm: React.PropTypes.func.isRequred
).isRequired
```

That way React will perform validation on the object.  Using `propTypes` in this way is nice since it's a dev-only check that's removed when compiled for production